src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx

```diff
@@ -111,17 +111,17 @@ export class DataSourceAggregatedView extends React.Component<
     let items = [];
 
     // only display active data sources
-    if (this.props.activeDataSourceIds && this.props.activeDataSourceIds.length > 0) {
-      items = this.props.activeDataSourceIds.map((id) => {
+    if (this.props.displayAllCompatibleDataSources) {
+      items = [...this.state.allDataSourcesIdToTitleMap.values()].map((title) => {
         return {
-          name: this.state.allDataSourcesIdToTitleMap.get(id),
+          name: title,
           disabled: true,
         };
       });
     } else {
-      items = [...this.state.allDataSourcesIdToTitleMap.values()].map((title) => {
+      items = this.props.activeDataSourceIds!.map((id) => {
```

Comment:
Force unwrap looks a little risky here. Are there any possibilities that activeDataSourceIds will be undefined?
Maybe we can update it in following way to avoid force unwrap:
```
const activeDataSoureceIds = this.props.activeDataSoruceIds ?? [];
items = activeDataSoureceIds.map() .....
```
What do you think?

Reply:
Right, sine the two states start to diverge, I think we can make it simpler by having a separate component for display active. The reason I am hesitate to give a default value is the activeDataSourceIds are required when displayAllCompatibleDataSources is false